### PR TITLE
feat(module: popconfirm): support async close

### PR DESCRIPTION
**Pull request by rorry121**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/popconfirm/popconfirm.ts [hunk 1]
```diff
@@ -268,8 +274,28 @@ export class NzPopconfirmComponent extends NzToolTipComponent implements OnDestr
...
+          finalize(() => {
+            this.confirmLoading = false;
+            this.cdr.markForCheck();
+          }),
+          takeUntil(this.nzVisibleChange),
+          takeUntil(this.destroy$)
+        )
+        .subscribe(value => {
+          if (value) {
+            this.nzOnConfirm.next();
```
Comment (hullis) [inline components/popconfirm/popconfirm.ts:291]: L291-292 & L296-297 could be extracted to a function.

**Review by hullis — COMMENTED** [review]
Thanks for your PR!

File: components/popconfirm/popconfirm.ts (hunk 1 above)
Comment (rorry121) [reply components/popconfirm/popconfirm.ts]: updated！

**Review by hullis — APPROVED** [review]
LGTM

**Pull request merged**